### DISCUSSION TITLE
User manual references nonexistent btGimpactTriangleMeshShape

### DISCUSSION THREAD

**Discussion by mast4461** [bulletphysics/bullet3#3643] [General]
The image on page 13 of the [user manual](https://github.com/bulletphysics/bullet3/blob/2.88/docs/Bullet_User_Manual.pdf) (at least versions 2.82 and 2.83) suggests to use `btGimpactTriangleMeshShape` to handle collisions for moving objects that can not be approximated as a single primitive, a convex hull of a triangle mesh, or a compound shape:

![image](https://user-images.githubusercontent.com/3101405/64117994-e9a1e300-cd96-11e9-8d92-d362ee608af6.png)

**There is [no occurrence](https://github.com/bulletphysics/bullet3/search?q=btGimpactTriangleMeshShape&unscoped_q=btGimpactTriangleMeshShape) of the word `btGimpactTriangleMeshShape`** in the current source code. There is however something called `btGImpactMeshShape`. Perhaps that would be the correct suggestion, although I'm not yet familiar enough with the source code to know that.

**Comment by jrz371** [comment]
Correct, the btGImpactMeshShape is what it should be.
The btGImpactMeshShape handles dynamic concave shapes. In order for it to work correctly you'll have to ensure you register the btGImpactCollisionAlgorithm. Be aware though that the GImpact stuff does not work well with softbody simulations.

**Comment by mast4461** [comment]
Thanks for the prompt reply and tips, @jrz371!
I'm trying to add support for dynamic concave mesh colliders to [ammo.js](https://github.com/kripken/ammo.js), a transpilation of Bullet to JavaScript, for a very specific use case where I don't necessarily need any realtime physics simulation, only on demand collision detection between kinematic rigid bodies with (possibly) concave mesh colliders, or possibly compound shapes consisting of convex meshes.
